compiler/rustc_ast/src/visitors.rs

```diff
@@ -554,6 +558,16 @@ macro_rules! make_ast_visitor {
             return_result!(V)
         }
 
+        pub fn walk_fn_decl<$($lt,)? V: $trait$(<$lt>)?>(
+            vis: &mut V,
+            decl: ref_t!(P!(FnDecl))
```

Comment:
Should this just take a `ref_t!(FnDecl)`?

Reply:
Yep. Done =)

Reply:
Removing `P!` here also allows the removal of `P!` on `visit_fn_decl` mentioned bellow!